node/src/components/storage.rs

```diff
@@ -307,6 +308,44 @@ impl Storage {
         drop(cursor);
         drop(block_txn);
 
+        info!("Checking block body db");
+        let body_txn = env.begin_ro_txn()?;
+        let mut cursor = body_txn.open_ro_cursor(block_body_db)?;
+
+        for (raw_key, raw_val) in cursor.iter() {
+            let body: BlockBody = lmdb_ext::deserialize(raw_val)?;
+            assert_eq!(
+                raw_key,
+                body.hash().as_ref(),
+                "found corrupt block body in database"
+            );
+        }
+        info!("block body db check complete");
+        drop(cursor);
+        drop(body_txn);
```

Comment:
Extract to a method please.

Reply:
@marc-casperlabs Any thoughts on how to extract this

Reply:
Extracted